paddle/fluid/operators/mkldnn/matmul_v2_mkldnn_op.cc

```diff
@@ -0,0 +1,199 @@
+
+      x_strides.reserve(x_dims.size());
+      y_strides.reserve(x_dims.size());
+      out_strides.reserve(x_dims.size());
+
+      if (!trans_x) {
+        x_strides.insert(x_strides.end(), {M * K, K, 1});
+      } else {
+        x_strides.insert(x_strides.end(), {M * K, 1, M});
+      }
+
+      if (!trans_y) {
+        y_strides.insert(y_strides.end(), {N * K, N, 1});
+      } else {
+        y_strides.insert(y_strides.end(), {N * K, 1, K});
+      }
+
+      out_strides.insert(out_strides.end(), {M * N, N, 1});
+      out_ddims.insert(out_ddims.end(),
+                       {std::max(x_dims[MB_idx], y_dims[MB_idx]), M, N});
+
+      for (int i = x_dims.size() - 4; i >= 0; --i) {
+        out_ddims[i] = std::max(x_dims[i], y_dims[i]);
+        x_strides[i] = x_dims[i + 1] * x_strides[i + 1];
+        y_strides[i] = y_dims[i + 1] * y_strides[i + 1];
+        out_strides[i] = out_ddims[i + 1] * out_strides[i + 1];
+      }
+
+      auto x_md = memory::desc(x_dims, MKLDNNGetDataType<T>(), x_strides);
+      auto y_md = memory::desc(y_dims, MKLDNNGetDataType<T>(), y_strides);
+      auto out_md =
+          memory::desc(out_ddims, MKLDNNGetDataType<T>(), out_strides);
+
+      this->AcquireForwardPrimitiveDescriptor(x_md, y_md, out_md);
+    }
+  }
+
+  std::shared_ptr<memory> AcquireWeightsMemory(const Tensor* input) {
+    const T* input_data = input->data<T>();
+    return this->AcquireMemoryFromPrimitive(this->fwd_pd_->weights_desc(),
+                                            to_void_cast<T>(input_data),
+                                            "@weights_mem_p");
+  }
+};
+
+template <typename T>
+class MatMulV2MKLDNNKernel : public framework::OpKernel<T> {
+ public:
+  void Compute(const ExecutionContext& ctx) const override { RunKernel(ctx); }
+
+ private:
+  void CalculateMatrixDims(const ExecutionContext& ctx,
+                           const std::vector<int64_t>& x_dims,
+                           const std::vector<int64_t>& y_dims,
+                           std::vector<int64_t>& x_bd_dims,
+                           std::vector<int64_t>& y_bd_dims,
+                           std::vector<int64_t>& out_dims, Tensor* out) const {
+    if (x_dims.size() == 1) {
+      x_bd_dims[x_bd_dims.size() - 1] = x_dims[0];
+    } else {
+      for (size_t i = 0; i < x_dims.size(); ++i) {
+        x_bd_dims[i] = x_dims[i];
+      }
+    }
+    if (y_dims.size() == 1) {
+      y_bd_dims[x_bd_dims.size() - 2] = y_dims[0];
+    } else {
+      for (size_t i = 0; i < y_dims.size(); ++i) {
+        y_bd_dims[i] = y_dims[i];
+      }
+    }
+
+    if ((y_dims.size() == x_dims.size()) && y_dims.size() > 2) {
+      for (size_t i = 0; i < x_dims.size() - 2; ++i) {
+        out_dims[i] = std::max(x_dims[i], y_dims[i]);
+      }
+      out->Resize(framework::make_ddim(out_dims));
+    }
+  }
+
+  void RunKernel(const ExecutionContext& ctx) const {
+    const auto& dev_ctx =
+        ctx.template device_context<platform::MKLDNNDeviceContext>();
```

Comment:
The namespace is not necessary because of using statement at the beginning of file.
```suggestion
        ctx.template device_context<MKLDNNDeviceContext>();
```